aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -1,130 +1,139 @@
 
-                elif type(each_block) in increment_for and type(each_block) not in nested_for:
-                    if isinstance(each_block, BinaryOperation):
-                        if each_block.operator in logical_operators:
-                            self.complexity += 1
-                            self._increment_logical_operators(each_block, each_block.operator)
+        elif each_block_type in increment_for and each_block_type in nested_for:
+            self.complexity += 1 + nested_level
+            self._traverse_childs(ast, each_block, nested_level + 1)
 
-                    elif isinstance(each_block, MethodInvocation):
-                        is_recursion = self._is_recursion_call(each_block)
-                        self.complexity += is_recursion
+        elif each_block_type in increment_for and each_block_type not in nested_for:
+            if ast.get_type(each_block) == ASTNodeType.BINARY_OPERATION:
+                bin_operator = ast.get_binary_operation_name(each_block)
+                if bin_operator in logical_operators:
+                    self.complexity += 1
+                    self._increment_logical_operators(ast, each_block)
 
-                    else:
-                        self.complexity += 1
-                        self._traverse_childs(each_block, nested_level)
+            elif each_block_type == ASTNodeType.METHOD_INVOCATION:
+                is_recursion = self._is_recursion_call(ast, each_block)
+                self.complexity += is_recursion
 
-                else:
-                    self._traverse_childs(each_block, nested_level)
+            else:
+                self.complexity += 1
+                self._traverse_childs(ast, each_block, nested_level)
+        else:
+            self._traverse_childs(ast, each_block, nested_level)
+
+    def _get_node_name(self, ast, node) -> str:  # type: ignore
+        extracted_name = None
+        names = list(ast.children_with_type(node, ASTNodeType.STRING))
+        for each_string in names:
+            method_name = ast.get_attr(each_string, 'string')
+            if not method_name.startswith('/') and re.search(r'[^\W\d]', method_name) is not None:
+                extracted_name = method_name
+                return extracted_name
 
     def value(self, filename: str) -> int:
-
-        tree = build_ast(filename)
-        for _, class_body in tree.filter(javalang.tree.ClassDeclaration):
-            for each_object in class_body.body:
-                if isinstance(each_object, MethodDeclaration):
-
-                    # memorize the name for detecting recursion call
-                    self.method_name = each_object.name
-                    self._get_complexity(each_object, 0)
+        p = JavaPackage(filename)
+        for class_name in p.java_classes:
+            tree = p.java_classes[class_name]
+
+            declareted_methods = tree.subtrees_with_root_type(ASTNodeType.METHOD_DECLARATION)
```

Comment:
it is more inconviniets, because those elements are sets. Hence, I need to refer, firstly values of dict and after initialize the loop to each element